linalg/solvers.cpp

```diff
@@ -70,6 +70,54 @@ double IterativeSolver::Dot(const Vector &x, const Vector &y) const
 
 void IterativeSolver::SetPrintLevel(int print_lvl)
 {
+#ifdef MFEM_USE_MPI
+   int rank = 0;
+   if (comm != MPI_COMM_NULL)
+   {
+      MPI_Comm_rank(comm, &rank);
+   }
+
+   if (rank == 0)
+   {
+      print_level = print_lvl;
+   }
+#endif
+
+   print_options = PrintOptions();
+   switch (print_level)
+   {
+      case -1:
+         break;
+
+      case 0:
+         print_options.errors = true;
+         print_options.warnings = true;
+         break;
+
+      case 1:
+         print_options.errors = true;
+         print_options.warnings = true;
+         print_options.iteration_details = true;
+         break;
+
+      case 2:
+         print_options.errors = true;
+         print_options.warnings = true;
+         print_options.summary = true;
+         break;
+
+      default:
+#ifdef MFEM_USE_MPI
+         if (rank == 0)
+#endif
+            MFEM_WARNING("Unknown print level " << print_level <<
+                         ". Defaulting to level 0.");
+
+         print_options.errors = true;
+         print_options.warnings = true;
+         break;
+   }
```

Comment:
Perhaps this logic can be moved to a `PrintOptions(int legacy_print_level)` constructor?

Reply:
Since I exchanged the struct with an enum I refactored this into a function.